### DISCUSSION TITLE
Is it possible to group CMS fields, but avoid nested frontmatter keys?

### DISCUSSION THREAD

**Discussion by jobee** [tinacms/tinacms#4027] [Help] [answered]
For better UX I'd like to hide certain CMS fields.

For example:

```typescript
export function advancedFields() {
  return [
    {
      type: "object",
      name: "advanced",
      label: "Advanced",
      fields: [
        {
          type: "string",
          name: "layout",
          label: "Layout",
        },
        {
          type: "string",
          name: "slug",
          label: "Slug",
          description: "to override current path segment (filename by default)",
        },
        {
          type: "string",
          name: "url",
          label: "URL",
          description: "to override the entire URL path",
        },
        {
          type: "string",
          name: "aliases",
          label: "Aliases",
          description: "provide URL paths to redirect to this page (quick solution, not recommended for SEO purposes)",
          list: true,
        },
      ],
    },
  ] as TinaField[];
}
```

Results in:

```yaml
advanced:
  layout: contact
  slug: new-slug
  aliases:
    - old-path
    - another-old-path
```

But my SSG Hugo expects:

```yaml
layout: contact
slug: new-slug
aliases:
  - old-path
  - another-old-path
```

Thank you for any input on this!

**Comment by jeffsee55** [comment, marked as answer]
Sorry but this isn't currently possible

**Reply by jobee** [reply]
Thanks for your quick reply.

Any plans to support this? Larger projects that rely on a certain front matter structure and have a lot of edit fields could benefit from a more flexible UI layout. But I completely understand that it's not a top priority.

**Reply by jeffsee55** [reply]
Definitely something I'd love to see as well, we've had requests for this in the past, I'm not sure of the best API for this, and there are a lot of moving parts with how our form field groups are handled. I wonder if it might be something as simple as using `..` to the beginning of the field names 🤔 

```
export function advancedFields() {
  return [
    {
      type: "object",
      name: "advanced",
      label: "Advanced",
      fields: [
        {
          type: "string",
          name: "..layout",
          label: "Layout",
        },
        {
          type: "string",
          name: "..slug",
          label: "Slug",
          description: "to override current path segment (filename by default)",
        },
        {
          type: "string",
          name: "..url",
          label: "URL",
          description: "to override the entire URL path",
        },
        {
          type: "string",
          name: "..aliases",
          label: "Aliases",
          description: "provide URL paths to redirect to this page (quick solution, not recommended for SEO purposes)",
          list: true,
        },
      ],
    },
  ] as TinaField[];
}
```

That's admittedly sort of strange, but... maybe makes sense? Would welcome some thoughts on API design here, but there's a fair bit of complexity in how we get the appropriate fields to show up.

**Reply by jobee** [reply]
If possible, I would probably prefer an absolute path rather than a relative one.

Something like:

`name: "/layout"` or `name: "frontmatter.layout"`

Side note: Reading the docs, I initially thought that `frontmatter.` overwrites the entire path in this example here:
https://tina.io/docs/reference/toolkit/fields/list/#with-the-object-object-field

---

And additionally force nested levels with a dot syntax, such as:

`name: "/layout.nested"` results in:

```yaml
---
layout:
  nested: "..."
---
```
